src/DataTypes/Serializations/SerializationSparse.cpp

```diff
@@ -152,7 +152,7 @@ void SerializationSparse::enumerateStreams(
     const StreamCallback & callback,
     const SubstreamData & data) const
 {
-    const auto * column_sparse = data.column ? &assert_cast<const ColumnSparse &>(*data.column) : nullptr;
+    const auto * column_sparse = data.column ? typeid_cast<const ColumnSparse *>(data.column.get()) : nullptr;
```

Comment:
In this PR column is provided in `enumerateStreams` in more places (for example in `MergeTreeDataPartWriter*`) because we need Dynamic column to know it's structure. And in case of Sparse serialization we use `ColumnSparse` only when we have certain ratio of defaults, so, in `SerializationSparse` sometimes column can be `ColumnSparse` and sometimes it can be nested column (you can also see this logic in `SerializationSparse::serializeBinaryBulkWithMultipleStreams`). So, here we also can have `ColumnSparse` or nested column (actually, I need to fix this place and use column futher in `enumerateStreams` if it's not nullpt and not `ColumnSparse`).